server/app/auth/CiviFormHttpActionAdapter.java

```diff
@@ -0,0 +1,35 @@
+package auth;
+
+import org.pac4j.core.context.HttpConstants;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.exception.http.HttpAction;
+import org.pac4j.play.PlayWebContext;
+import org.pac4j.play.http.PlayHttpActionAdapter;
+import play.mvc.Result;
+import play.mvc.Results;
+
+/**
+ * The action adapter allows pac4j-consuming code to intervene and change the default behavior of
+ * handling a given HTTP result. The default {@link PlayHttpActionAdapter} does not provide a way of
+ * customizing behavior beyond mapping status codes to {@link Result} objects. In CiviForm's case,
+ * we want to handle unauthorized requests to the API differently, since there is not a UX
```

Comment:
Suggestion: write the doc with the WHAT first and WHY second.

This reads as a justification for it's existence primarily without a lot of clear explanation what its actually doing.

Reply:
done